spk/transmission/src/service-setup.sh

```diff
@@ -14,119 +14,55 @@ TRANSMISSION="${SYNOPKG_PKGDEST}/bin/transmission-daemon"
 
 SERVICE_COMMAND="${TRANSMISSION} -g ${SYNOPKG_PKGVAR} -x ${PID_FILE} -e ${LOG_FILE}"
 
-validate_preinst ()
-{
-    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ] && [ "${SYNOPKG_DSM_VERSION_MAJOR}" -lt 7 ]; then
-        # If chosen, they need to exist
-        if [ -n "${wizard_watch_dir}" ] && [ ! -d "${wizard_watch_dir}" ]; then
-            echo "Watch directory ${wizard_watch_dir} does not exist."
-            exit 1
-        fi
-        if [ -n "${wizard_incomplete_dir}" ] && [ ! -d "${wizard_incomplete_dir}" ]; then
-            echo "Incomplete directory ${wizard_incomplete_dir} does not exist."
-            exit 1
-        fi
-    fi
-
-    exit 0
-}
-
 service_postinst ()
 {
-    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
-        # Attempt to create the folders after the user/group/shared folder has been created by the package (DSM7+)
-        # Let the package manage the "watch" and "incomplete" folders for DSM7
-        # https://github.com/SynoCommunity/spksrc/issues/4766#issuecomment-899875151
-        if [ "${SYNOPKG_DSM_VERSION_MAJOR}" -ge 7 ]; then
-            wizard_watch_dir="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/watch"
-            wizard_incomplete_dir="${wizard_volume:=/volume1}/${wizard_download_dir:=/downloads}/incomplete"
-            mkdir -p "${wizard_watch_dir}"
-            mkdir -p "${wizard_incomplete_dir}"
-        else
-            # Set permissions for optional folders for DSM <7
-            # existance is validated with validate_preinst
-            set_syno_permissions "${wizard_watch_dir}" "${GROUP}"
-            set_syno_permissions "${wizard_incomplete_dir}" "${GROUP}"
+    if [ "${SYNOPKG_PKG_STATUS}" = "INSTALL" ]; then
+        # Capture wizard variable
+        TXN_DNLOAD=${wizard_download_dir:=volume1/downloads}
+        # Check that the path exists, if not use path in package shares
+        if [ ! -d "${TXN_DNLOAD}" ]; then
+            TXN_DNLOAD=$(realpath "/var/packages/${SYNOPKG_PKGNAME}/shares/${wizard_download_share}")
         fi
+        TXN_FUNCTS=("complete" "incomplete" "watch")
+        TXN_FOLDRS=("complete" "incomplete" "watch-transmission")
+        TXN_PATHS=()
```

Comment:
You are correct that we shouldn't be manipulating existing setups on upgrade. This was something that @tothf was very clear on in his note https://github.com/SynoCommunity/spksrc/issues/5595#issuecomment-1470307364. I actually agreed with this perspective which is why all the upgrade scripts were removed. These folders are only created on first install of a new deployment. As this is wrapped in a `SYNOPKG_PKG_STATUS` check for `INSTALL` this will not fire for upgrades.

Reply:
My bad !

Indeed it's fine for a simple install.